File: app/src/main/java/com/itmoldova/db/ArticleDao.java
package com.itmoldova.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.itmoldova.model.Article;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface ArticleDao {

    @Query("select * from Article")
    Flowable<List<Article>> loadAllArticles();

    @Query("select * from Article where guid = :id")
    Flowable<Article> getArticleById(String id);

    @Delete
    void deleteArticle(Article article);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertArticle(Article article);
}

File: app/src/main/java/com/itmoldova/db/ArticleDao.kt
package com.itmoldova.db

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.itmoldova.model.Article

import io.reactivex.Flowable

@Dao
interface ArticleDao {

    @Query("select * from Article")
    fun loadAllArticles(): Flowable<List<Article>>

    @Query("select * from Article where guid = :id")
    fun getArticleById(id: String): Flowable<Article>

    @Delete
    fun deleteArticle(article: Article)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertArticle(article: Article)
}
